app/src/main/kotlin/com/mapbox/vision/teaser/recorder/RecorderFragment.kt

```diff
@@ -20,20 +26,30 @@ class RecorderFragment : Fragment() {
         private fun buildFileName() = dateFormat.format(Date(System.currentTimeMillis()))
 
         private const val ARG_PARAM_SESSIONS_PATH = "ARG_PARAM_SESSIONS_PATH"
+		private const val ARG_PARAM_JSON_ROUTE = "ARG_PARAM_JSON_ROUTE"
         val TAG: String = RecorderFragment::class.java.simpleName
 
-        fun newInstance(sessionsPath: String) = RecorderFragment().apply {
+        fun newInstance(sessionsPath: String, jsonRoute: String?) = RecorderFragment().apply {
             arguments = Bundle().apply {
                 putString(ARG_PARAM_SESSIONS_PATH, sessionsPath)
+				putString(ARG_PARAM_JSON_ROUTE, jsonRoute)
             }
         }
     }
 
-    private var baseSessionsPath: String? = ""
+    private lateinit var baseSessionsPath: String
+    private var routePoints: Array<RoutePoint> = arrayOf()
```

Comment:
```suggestion
    private val routePoints: Array<RoutePoint> = mutableArrayOf()
```

Reply:
there is no muttableArrayOf(), only reassign array

Reply:
Nevermind then.